{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-02T20:36:18.534Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-02T21:39:40.796Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-02T21:40:33.167Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-02T21:40:54.505Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-02T21:41:18.182Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-02T21:41:49.142Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-02T21:42:41.154Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-02T21:43:59.445Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T15:59:19.815Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T15:59:48.754Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:00:56.708Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:01:10.063Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:01:35.155Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:02:44.016Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:03:09.856Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:07:43.684Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:08:14.360Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:08:44.521Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:09:26.904Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:10:01.963Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:10:26.438Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:10:44.900Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:11:25.751Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:11:59.763Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:14:26.268Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:14:42.896Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:16:26.567Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T16:17:34.066Z"}
{"level":"error","message":"Email service connection failed: Connection closed","stack":"Error: Connection closed\n    at SMTPConnection.<anonymous> (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\nodemailer\\lib\\smtp-transport\\index.js:355:33)\n    at Object.onceWrapper (node:events:633:28)\n    at SMTPConnection.emit (node:events:519:28)\n    at SMTPConnection.emit (node:domain:488:12)\n    at SMTPConnection._destroy (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\nodemailer\\lib\\smtp-connection\\index.js:897:14)\n    at SMTPConnection._onClose (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\nodemailer\\lib\\smtp-connection\\index.js:866:14)\n    at TLSSocket.SMTPConnection._onSocketClose (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\nodemailer\\lib\\smtp-connection\\index.js:197:42)\n    at Object.onceWrapper (node:events:634:26)\n    at TLSSocket.emit (node:events:531:35)\n    at TLSSocket.emit (node:domain:488:12)","timestamp":"2025-02-03T21:11:52.197Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T21:11:52.199Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T21:11:53.762Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T21:25:59.328Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T21:26:22.545Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T21:26:47.994Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T21:27:47.341Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-03T23:20:43.167Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/buy-data on this server","method":"POST","path":"/api/v1/buy-data","stack":"Error: Can't find /api/v1/buy-data on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:81:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-04T21:36:43.301Z"}
{"level":"error","message":"Cannot read properties of undefined (reading 'plan_type')","timestamp":"2025-02-04T21:40:54.272Z"}
{"error":{},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'plan_type')","method":"POST","path":"/api/v1/buy-data","stack":"TypeError: Cannot read properties of undefined (reading 'plan_type')\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:43:19)\n    at DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:44)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:175:3)","timestamp":"2025-02-04T21:40:54.287Z"}
{"level":"error","message":"Cannot read properties of undefined (reading 'filter')","timestamp":"2025-02-04T21:42:02.396Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'filter')","method":"POST","path":"/api/v1/buy-data","stack":"Error: Cannot read properties of undefined (reading 'filter')\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:61:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-04T21:42:02.406Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-04T21:42:02.796Z"}
{"level":"error","message":"Cannot read properties of undefined (reading 'filter')","timestamp":"2025-02-04T21:44:42.638Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'filter')","method":"POST","path":"/api/v1/buy-data","stack":"Error: Cannot read properties of undefined (reading 'filter')\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:63:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-04T21:44:42.646Z"}
{"level":"error","message":"Cannot read properties of undefined (reading 'filter')","timestamp":"2025-02-04T21:46:26.857Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'filter')","method":"POST","path":"/api/v1/buy-data","stack":"Error: Cannot read properties of undefined (reading 'filter')\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:62:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-04T21:46:26.865Z"}
{"level":"error","message":"Cannot read properties of undefined (reading 'filter')","timestamp":"2025-02-04T21:47:50.567Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'filter')","method":"POST","path":"/api/v1/buy-data","stack":"Error: Cannot read properties of undefined (reading 'filter')\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:62:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-04T21:47:50.573Z"}
{"level":"error","message":"Cannot read properties of undefined (reading 'filter')","timestamp":"2025-02-04T21:48:21.911Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'filter')","method":"POST","path":"/api/v1/buy-data","stack":"Error: Cannot read properties of undefined (reading 'filter')\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:62:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-04T21:48:21.922Z"}
{"level":"error","message":"Cannot read properties of undefined (reading 'filter')","timestamp":"2025-02-04T21:49:08.051Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'filter')","method":"POST","path":"/api/v1/buy-data","stack":"Error: Cannot read properties of undefined (reading 'filter')\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:62:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-04T21:49:08.058Z"}
{"level":"error","message":"Cannot read properties of undefined (reading 'filter')","timestamp":"2025-02-04T21:50:06.785Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'filter')","method":"POST","path":"/api/v1/buy-data","stack":"Error: Cannot read properties of undefined (reading 'filter')\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:62:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-04T21:50:06.791Z"}
{"level":"error","message":"Cannot read properties of undefined (reading 'filter')","timestamp":"2025-02-04T21:58:03.743Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'filter')","method":"POST","path":"/api/v1/buy-data","stack":"Error: Cannot read properties of undefined (reading 'filter')\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:64:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-04T21:58:03.751Z"}
{"level":"error","message":"response[plan_network].filter is not a function","timestamp":"2025-02-04T21:59:53.639Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] response[plan_network].filter is not a function","method":"POST","path":"/api/v1/buy-data","stack":"Error: response[plan_network].filter is not a function\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:64:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-04T21:59:53.668Z"}
{"level":"error","message":"networkServices.filter is not a function","timestamp":"2025-02-04T22:02:27.784Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] networkServices.filter is not a function","method":"POST","path":"/api/v1/buy-data","stack":"Error: networkServices.filter is not a function\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:75:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-04T22:02:27.791Z"}
{"level":"error","message":"networkServices.filter is not a function","timestamp":"2025-02-04T22:03:21.362Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] networkServices.filter is not a function","method":"POST","path":"/api/v1/buy-data","stack":"Error: networkServices.filter is not a function\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:76:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-04T22:03:21.372Z"}
{"level":"error","message":"networkServices.filter is not a function","timestamp":"2025-02-05T00:58:19.450Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] networkServices.filter is not a function","method":"POST","path":"/api/v1/buy-data","stack":"Error: networkServices.filter is not a function\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:75:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-05T00:58:19.462Z"}
{"level":"error","message":"Failed to find data plan","timestamp":"2025-02-05T01:01:52.636Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Failed to find data plan","method":"POST","path":"/api/v1/buy-data","stack":"Error: Failed to find data plan\n    at DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:19:23)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-02-05T01:01:52.644Z"}
{"level":"error","message":"Failed to find data plan","timestamp":"2025-02-05T01:03:17.555Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Failed to find data plan","method":"POST","path":"/api/v1/buy-data","stack":"Error: Failed to find data plan\n    at DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:19:23)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-02-05T01:03:17.566Z"}
{"level":"error","message":"Failed to find data plan","timestamp":"2025-02-05T01:04:14.628Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Failed to find data plan","method":"POST","path":"/api/v1/buy-data","stack":"Error: Failed to find data plan\n    at DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:19:23)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-02-05T01:04:14.633Z"}
{"level":"error","message":"timeout of 10000ms exceeded","timestamp":"2025-02-05T01:07:14.338Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] timeout of 10000ms exceeded","method":"POST","path":"/api/v1/buy-data","stack":"Error: timeout of 10000ms exceeded\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:78:19)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-05T01:07:14.357Z"}
{"level":"error","message":"timeout of 10000ms exceeded","timestamp":"2025-02-05T01:07:40.256Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] timeout of 10000ms exceeded","method":"POST","path":"/api/v1/buy-data","stack":"Error: timeout of 10000ms exceeded\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:78:19)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-05T01:07:40.261Z"}
{"level":"error","message":"Cannot read properties of undefined (reading 'filter')","timestamp":"2025-02-05T01:19:20.449Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'filter')","method":"POST","path":"/api/v1/buy-data","stack":"Error: Cannot read properties of undefined (reading 'filter')\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:76:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-05T01:19:20.463Z"}
{"level":"error","message":"Cannot read properties of undefined (reading 'filter')","timestamp":"2025-02-05T01:20:10.449Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'filter')","method":"POST","path":"/api/v1/buy-data","stack":"Error: Cannot read properties of undefined (reading 'filter')\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:76:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-05T01:20:10.456Z"}
{"level":"error","message":"No matching service found","timestamp":"2025-02-05T01:20:42.398Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] No matching service found","method":"POST","path":"/api/v1/buy-data","stack":"Error: No matching service found\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:76:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-05T01:20:42.402Z"}
{"level":"error","message":"No matching service found","timestamp":"2025-02-05T01:21:18.953Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] No matching service found","method":"POST","path":"/api/v1/buy-data","stack":"Error: No matching service found\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:76:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-05T01:21:18.962Z"}
{"level":"error","message":"socket hang up","timestamp":"2025-02-05T01:40:09.131Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] socket hang up","method":"POST","path":"/api/v1/buy-data","stack":"Error: socket hang up\n    at GladTidingsService.findData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\gladtidings.ts:76:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async DataCntroller.buyData (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\dataController.ts:17:30)","timestamp":"2025-02-05T01:40:09.149Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-05T01:47:19.430Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-09T07:38:03.996Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-09T07:39:25.204Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-09T07:44:20.847Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-09T07:44:59.393Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-09T07:51:48.013Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-09T07:52:16.339Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-09T08:04:59.301Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-09T10:18:06.554Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-09T10:18:43.518Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-09T11:00:11.951Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/card-payment on this server","method":"POST","path":"/api/v1/card-payment","stack":"Error: Can't find /api/v1/card-payment on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:81:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-09T11:15:08.016Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/card-payment on this server","method":"POST","path":"/api/v1/card-payment","stack":"Error: Can't find /api/v1/card-payment on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:81:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-09T11:18:05.818Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/card-payment on this server","method":"POST","path":"/api/v1/card-payment","stack":"Error: Can't find /api/v1/card-payment on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:81:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-09T11:21:37.207Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-09T11:59:35.228Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-10T08:52:15.139Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-10T08:52:44.978Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-10T08:55:23.514Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-10T08:55:43.934Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-10T08:56:25.607Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-10T08:56:40.573Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-10T08:58:15.087Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-10T09:06:55.977Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-10T09:07:07.721Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-10T09:07:34.116Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-10T09:07:45.673Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-10T09:10:12.126Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-10T09:10:17.314Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-10T09:11:00.736Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-14T23:11:23.558Z"}
{"level":"error","message":"Email service connection failed: Connection closed","stack":"Error: Connection closed\n    at SMTPConnection.<anonymous> (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\nodemailer\\lib\\smtp-transport\\index.js:355:33)\n    at Object.onceWrapper (node:events:633:28)\n    at SMTPConnection.emit (node:events:519:28)\n    at SMTPConnection.emit (node:domain:488:12)\n    at SMTPConnection._destroy (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\nodemailer\\lib\\smtp-connection\\index.js:897:14)\n    at SMTPConnection._onClose (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\nodemailer\\lib\\smtp-connection\\index.js:866:14)\n    at TLSSocket.SMTPConnection._onSocketClose (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\nodemailer\\lib\\smtp-connection\\index.js:197:42)\n    at Object.onceWrapper (node:events:634:26)\n    at TLSSocket.emit (node:events:531:35)\n    at TLSSocket.emit (node:domain:488:12)","timestamp":"2025-02-14T23:11:24.548Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-14T23:11:24.550Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-14T23:17:24.520Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-15T00:04:44.300Z"}
{"error":{"ErrorCodes":"PIN required","isOperational":true,"statusCode":401},"level":"error","message":"[Global Error Handler] Unauthorized","method":"POST","path":"/api/v1/admin/post-price","stack":"Error: Unauthorized\n    at AuthController.isAdmin (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:605:13)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:239:9\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:261:12\n    at getSecret (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:97:14)\n    at Object.module.exports [as verify] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:101:10)\n    at authenticationToken (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:230:11)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)","timestamp":"2025-02-17T22:45:39.457Z"}
{"error":"Unauthorized","level":"error","message":"[Global Error Handler] undefined","method":"POST","path":"/api/v1/admin/post-price","timestamp":"2025-02-17T22:48:22.287Z"}
{"error":{"ErrorCodes":"PIN required","isOperational":true,"statusCode":401},"level":"error","message":"[Global Error Handler] Unauthorized","method":"POST","path":"/api/v1/admin/post-price","stack":"Error: Unauthorized\n    at AuthController.isAdmin (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:605:13)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:239:9\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:261:12\n    at getSecret (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:97:14)\n    at Object.module.exports [as verify] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:101:10)\n    at authenticationToken (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:230:11)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)","timestamp":"2025-02-17T22:48:42.760Z"}
{"error":{"message":"Unauthorized"},"level":"error","message":"[Global Error Handler] Unauthorized","method":"POST","path":"/api/v1/admin/post-price","timestamp":"2025-02-17T22:50:49.003Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server","method":"GET","path":"/api/v1/admin/get-prices/","stack":"Error: Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:87:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-17T23:23:45.963Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server","method":"GET","path":"/api/v1/admin/get-prices/","stack":"Error: Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:87:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-17T23:24:04.028Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server","method":"GET","path":"/api/v1/admin/get-prices/","stack":"Error: Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:87:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-17T23:28:40.064Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/admin/get-prices/?networkProvider=mtn on this server","method":"GET","path":"/api/v1/admin/get-prices/","stack":"Error: Can't find /api/v1/admin/get-prices/?networkProvider=mtn on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:87:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-17T23:30:47.435Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server","method":"GET","path":"/api/v1/admin/get-prices/","stack":"Error: Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:87:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-17T23:44:32.687Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server","method":"GET","path":"/api/v1/admin/get-prices/","stack":"Error: Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:87:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-17T23:47:13.586Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server","method":"GET","path":"/api/v1/admin/get-prices/","stack":"Error: Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:87:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-17T23:48:05.981Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-17T23:48:12.834Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server","method":"GET","path":"/api/v1/admin/get-prices/","stack":"Error: Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:87:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-17T23:50:01.262Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/admin/get-prices on this server","method":"GET","path":"/api/v1/admin/get-prices","stack":"Error: Can't find /api/v1/admin/get-prices on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:87:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-17T23:50:57.075Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/admin/get-prices?networkProvider=MTN on this server","method":"GET","path":"/api/v1/admin/get-prices","stack":"Error: Can't find /api/v1/admin/get-prices?networkProvider=MTN on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:87:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-17T23:51:08.855Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server","method":"GET","path":"/api/v1/admin/get-prices/","stack":"Error: Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:87:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-17T23:58:51.523Z"}
{"code":"EDNS","command":"CONN","errno":-3008,"hostname":"smtp.gmail.com","level":"error","message":"Email service connection failed: getaddrinfo ENOTFOUND smtp.gmail.com","stack":"Error: getaddrinfo ENOTFOUND smtp.gmail.com\n    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:118:26)","syscall":"getaddrinfo","timestamp":"2025-02-18T00:10:18.002Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-18T00:10:18.004Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/admin/get-prices?networkProvider=MTN on this server","method":"GET","path":"/api/v1/admin/get-prices","stack":"Error: Can't find /api/v1/admin/get-prices?networkProvider=MTN on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:87:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-18T00:16:25.074Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server","method":"GET","path":"/api/v1/admin/get-prices/","stack":"Error: Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:87:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-18T00:16:36.048Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server","method":"GET","path":"/api/v1/admin/get-prices/","stack":"Error: Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:87:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-18T00:22:32.308Z"}
{"error":{"isOperational":true,"statusCode":404},"level":"error","message":"[Global Error Handler] Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server","method":"GET","path":"/api/v1/admin/get-prices/","stack":"Error: Can't find /api/v1/admin/get-prices/?networkProvider=MTN on this server\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\index.ts:87:8\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at trim_prefix (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:328:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:286:9\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at Immediate.next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Immediate._onImmediate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:646:15)\n    at processImmediate (node:internal/timers:480:21)","timestamp":"2025-02-18T00:22:54.068Z"}
{"error":{},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'split')","method":"GET","path":"/api/v1/admin/get-data/MTN","stack":"TypeError: Cannot read properties of undefined (reading 'split')\n    at authenticationToken (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:221:32)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)","timestamp":"2025-02-18T23:13:12.391Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-18T23:13:23.878Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-18T23:23:49.878Z"}
{"code":"ETIMEDOUT","command":"CONN","level":"error","message":"Email service connection failed: Connection timeout","stack":"Error: Connection timeout\n    at SMTPConnection._formatError (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\nodemailer\\lib\\smtp-connection\\index.js:807:19)\n    at SMTPConnection._onError (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\nodemailer\\lib\\smtp-connection\\index.js:793:20)\n    at Timeout.<anonymous> (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\nodemailer\\lib\\smtp-connection\\index.js:237:22)\n    at listOnTimeout (node:internal/timers:573:17)\n    at processTimers (node:internal/timers:514:7)","timestamp":"2025-02-18T23:34:33.621Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-18T23:34:33.623Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-18T23:35:15.348Z"}
{"code":"EDNS","command":"CONN","errno":-3008,"hostname":"smtp.gmail.com","level":"error","message":"Email service connection failed: getaddrinfo ENOTFOUND smtp.gmail.com","stack":"Error: getaddrinfo ENOTFOUND smtp.gmail.com\n    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:118:26)","syscall":"getaddrinfo","timestamp":"2025-02-18T23:35:15.382Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-18T23:35:15.383Z"}
{"error":{},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'split')","method":"POST","path":"/api/v1/make-payment","stack":"TypeError: Cannot read properties of undefined (reading 'split')\n    at authenticationToken (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:219:32)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:47:12)","timestamp":"2025-02-19T07:13:08.333Z"}
{"details":{"amount":500,"customer":"stephen@ikhane.com"},"error":{"code":"ERR_INVALID_URL","input":"/api/v1/merchant/transactions/init-transaction"},"level":"error","message":"Payment initiation failed","timestamp":"2025-02-19T07:13:45.390Z"}
{"level":"error","message":"Payment initiation failed: Invalid URL","timestamp":"2025-02-19T07:13:45.393Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Payment initiation failed: Invalid URL","method":"POST","path":"/api/v1/make-payment","stack":"Error: Payment initiation failed: Invalid URL\n    at MonifyService.initiatePayment (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\payment.ts:179:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async PaymentController.initializePayment (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:38:23)","timestamp":"2025-02-19T07:13:45.399Z"}
{"level":"error","message":"Transaction validation failed: reference: Transaction reference is required","timestamp":"2025-02-19T07:46:07.650Z"}
{"error":{"_message":"Transaction validation failed","errors":{"reference":{"kind":"required","message":"Transaction reference is required","name":"ValidatorError","path":"reference","properties":{"message":"Transaction reference is required","path":"reference","type":"required"}}},"message":"Transaction validation failed: reference: Transaction reference is required","name":"ValidationError"},"level":"error","message":"[Global Error Handler] Transaction validation failed: reference: Transaction reference is required","method":"POST","path":"/api/v1/make-payment","stack":"ValidationError: Transaction validation failed: reference: Transaction reference is required\n    at model.Document.invalidate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongoose\\lib\\document.js:3334:32)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongoose\\lib\\document.js:3095:17\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongoose\\lib\\schemaType.js:1407:9\n    at processTicksAndRejections (node:internal/process/task_queues:77:11)","timestamp":"2025-02-19T07:46:07.669Z"}
{"level":"error","message":"Transaction validation failed: reference: Transaction reference is required","timestamp":"2025-02-19T07:46:38.173Z"}
{"error":{"_message":"Transaction validation failed","errors":{"reference":{"kind":"required","message":"Transaction reference is required","name":"ValidatorError","path":"reference","properties":{"message":"Transaction reference is required","path":"reference","type":"required"}}},"message":"Transaction validation failed: reference: Transaction reference is required","name":"ValidationError"},"level":"error","message":"[Global Error Handler] Transaction validation failed: reference: Transaction reference is required","method":"POST","path":"/api/v1/make-payment","stack":"ValidationError: Transaction validation failed: reference: Transaction reference is required\n    at model.Document.invalidate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongoose\\lib\\document.js:3334:32)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongoose\\lib\\document.js:3095:17\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongoose\\lib\\schemaType.js:1407:9\n    at processTicksAndRejections (node:internal/process/task_queues:77:11)","timestamp":"2025-02-19T07:46:38.181Z"}
{"level":"error","message":"Transaction validation failed: reference: Transaction reference is required","timestamp":"2025-02-19T07:47:29.888Z"}
{"error":{"_message":"Transaction validation failed","errors":{"reference":{"kind":"required","message":"Transaction reference is required","name":"ValidatorError","path":"reference","properties":{"message":"Transaction reference is required","path":"reference","type":"required"}}},"message":"Transaction validation failed: reference: Transaction reference is required","name":"ValidationError"},"level":"error","message":"[Global Error Handler] Transaction validation failed: reference: Transaction reference is required","method":"POST","path":"/api/v1/make-payment","stack":"ValidationError: Transaction validation failed: reference: Transaction reference is required\n    at model.Document.invalidate (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongoose\\lib\\document.js:3334:32)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongoose\\lib\\document.js:3095:17\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongoose\\lib\\schemaType.js:1407:9\n    at processTicksAndRejections (node:internal/process/task_queues:77:11)","timestamp":"2025-02-19T07:47:29.891Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-19T07:48:50.820Z"}
{"level":"error","message":"Cannot read properties of undefined (reading 'id')","timestamp":"2025-02-21T07:09:37.018Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T07:09:37.020Z"}
{"level":"error","message":"Cannot read properties of undefined (reading 'id')","timestamp":"2025-02-21T07:10:09.901Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T07:10:09.902Z"}
{"error":{},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'split')","method":"POST","path":"/api/v1/make-payment","stack":"TypeError: Cannot read properties of undefined (reading 'split')\n    at authenticationToken (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:219:32)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:47:12)","timestamp":"2025-02-21T07:10:18.402Z"}
{"level":"error","message":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: \"123031klsadked\" }","timestamp":"2025-02-21T07:10:24.171Z"}
{"error":{"code":11000,"errorResponse":{"code":11000,"errmsg":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: \"123031klsadked\" }","index":0,"keyPattern":{"reference":1},"keyValue":{"reference":"123031klsadked"}},"index":0,"keyPattern":{"reference":1},"keyValue":{"reference":"123031klsadked"}},"level":"error","message":"[Global Error Handler] E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: \"123031klsadked\" }","method":"POST","path":"/api/v1/make-payment","stack":"MongoServerError: E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: \"123031klsadked\" }\n    at InsertOneOperation.execute (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\insert.ts:88:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async tryOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:278:14)\n    at async executeOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:112:12)\n    at async Collection.insertOne (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\collection.ts:284:12)","timestamp":"2025-02-21T07:10:24.188Z"}
{"level":"error","message":"Cannot read properties of undefined (reading 'id')","timestamp":"2025-02-21T07:18:09.819Z"}
{"error":{},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'id')","method":"POST","path":"/api/v1/verify-payment","stack":"TypeError: Cannot read properties of undefined (reading 'id')\n    at verifyTransaction (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:92:24)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-02-21T07:18:09.824Z"}
{"level":"error","message":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: \"PAY_1738405391803_fa8966d4\" }","timestamp":"2025-02-21T07:18:42.508Z"}
{"error":{"code":11000,"errorResponse":{"code":11000,"errmsg":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: \"PAY_1738405391803_fa8966d4\" }","index":0,"keyPattern":{"reference":1},"keyValue":{"reference":"PAY_1738405391803_fa8966d4"}},"index":0,"keyPattern":{"reference":1},"keyValue":{"reference":"PAY_1738405391803_fa8966d4"}},"level":"error","message":"[Global Error Handler] E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: \"PAY_1738405391803_fa8966d4\" }","method":"POST","path":"/api/v1/verify-payment","stack":"MongoServerError: E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: \"PAY_1738405391803_fa8966d4\" }\n    at InsertOneOperation.execute (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\insert.ts:88:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async tryOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:278:14)\n    at async executeOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:112:12)\n    at async Collection.insertOne (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\collection.ts:284:12)","timestamp":"2025-02-21T07:18:42.519Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T13:31:59.434Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T13:46:25.832Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T16:26:36.000Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T16:38:16.950Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T16:39:35.059Z"}
{"level":"error","message":"Cannot read properties of undefined (reading '0')","timestamp":"2025-02-21T16:40:41.424Z"}
{"error":{},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading '0')","method":"POST","path":"/api/v1/verify-payment","stack":"TypeError: Cannot read properties of undefined (reading '0')\n    at verifyTransaction (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:77:59)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:237:9\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:261:12\n    at getSecret (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:97:14)\n    at Object.module.exports [as verify] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:101:10)\n    at authenticationToken (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:228:11)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)","timestamp":"2025-02-21T16:40:41.428Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T16:40:49.434Z"}
{"level":"error","message":"Cannot read properties of undefined (reading '0')","timestamp":"2025-02-21T16:41:26.037Z"}
{"error":{},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading '0')","method":"POST","path":"/api/v1/verify-payment","stack":"TypeError: Cannot read properties of undefined (reading '0')\n    at verifyTransaction (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:77:59)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:237:9\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:261:12\n    at getSecret (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:97:14)\n    at Object.module.exports [as verify] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:101:10)\n    at authenticationToken (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:228:11)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)","timestamp":"2025-02-21T16:41:26.045Z"}
{"level":"error","message":"transactionReference.tostring is not a function","timestamp":"2025-02-21T16:41:47.074Z"}
{"error":{},"level":"error","message":"[Global Error Handler] transactionReference.tostring is not a function","method":"POST","path":"/api/v1/verify-payment","stack":"TypeError: transactionReference.tostring is not a function\n    at verifyTransaction (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:76:40)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:237:9\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:261:12\n    at getSecret (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:97:14)\n    at Object.module.exports [as verify] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:101:10)\n    at authenticationToken (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:228:11)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)","timestamp":"2025-02-21T16:41:47.078Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T16:41:55.055Z"}
{"level":"error","message":"Cannot read properties of undefined (reading '0')","timestamp":"2025-02-21T16:42:04.236Z"}
{"error":{},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading '0')","method":"POST","path":"/api/v1/verify-payment","stack":"TypeError: Cannot read properties of undefined (reading '0')\n    at verifyTransaction (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:77:59)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:237:9\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:261:12\n    at getSecret (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:97:14)\n    at Object.module.exports [as verify] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\jsonwebtoken\\verify.js:101:10)\n    at authenticationToken (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:228:11)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)","timestamp":"2025-02-21T16:42:04.239Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T17:07:04.034Z"}
{"error":{"code":"ERR_BAD_REQUEST","config":{"adapter":["xhr","http","fetch"],"baseURL":"https://sandbox.monnify.com","env":{},"headers":{"Accept":"application/json, text/plain, */*","Accept-Encoding":"gzip, compress, deflate, br","Authorization":"Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsibW9ubmlmeS12YWx1ZS1hZGRlZC1zZXJ2aWNlIiwibW9ubmlmeS1wYXltZW50LWVuZ2luZSIsIm1vbm5pZnktZGlzYnVyc2VtZW50LXNlcnZpY2UiLCJtb25uaWZ5LW9mZmxpbmUtcGF5bWVudC1zZXJ2aWNlIl0sInNjb3BlIjpbInByb2ZpbGUiXSwiZXhwIjoxNzQwMTYxMDE3LCJhdXRob3JpdGllcyI6WyJNUEVfTUFOQUdFX0xJTUlUX1BST0ZJTEUiLCJNUEVfVVBEQVRFX1JFU0VSVkVEX0FDQ09VTlQiLCJNUEVfSU5JVElBTElaRV9QQVlNRU5UIiwiTVBFX1JFU0VSVkVfQUNDT1VOVCIsIk1QRV9DQU5fUkVUUklFVkVfVFJBTlNBQ1RJT04iLCJNUEVfUkVUUklFVkVfUkVTRVJWRURfQUNDT1VOVCIsIk1QRV9ERUxFVEVfUkVTRVJWRURfQUNDT1VOVCIsIk1QRV9SRVRSSUVWRV9SRVNFUlZFRF9BQ0NPVU5UX1RSQU5TQUNUSU9OUyJdLCJqdGkiOiJmYWQ2ODE4ZS03NDQ1LTRkY2UtOWFjYy02NjQxMGRmNzBlZWYiLCJjbGllbnRfaWQiOiJNS19URVNUXzA2UDk5RlRONzEifQ.PrUvXTWxmQYEQsAd1oaRSRfUdhsDufBqpvVM7OEMfcR1NZisxpLns51CNx9QMoZjlRLVbNtDyN9NVBAajbZsKT0Op30IeLeJJ45dOwyaNCQoCkZtHDcbbHDVNRd7uwjjyG37cIxZvS4DDERXEVAUz_V-ONa0spRWHTmbJKyTLQyzuyfgRBTwgycyFKITbFUVC9QKkdy9HZTAKeC47zbIThyve4gLh8_jayIHnQa6tqZ6cTu0ZbCLsuaD9GywQ4yqiPwPxKlblPYOSCutYpuhPVKescrDGbVgcv0SN5Y-1tm77OjvNfPP2Euz00ZtkGo9ilbBo90PkMHgrvk8wEJvxg","Content-Type":"application/json","User-Agent":"axios/1.7.9"},"maxBodyLength":-1,"maxContentLength":-1,"method":"get","timeout":10000,"transformRequest":[null],"transformResponse":[null],"transitional":{"clarifyTimeoutError":false,"forcedJSONParsing":true,"silentJSONParsing":true},"url":"/api/v2/transactions/MNFY%7C57%7C20250201112313%7C00000","xsrfCookieName":"XSRF-TOKEN","xsrfHeaderName":"X-XSRF-TOKEN"},"message":"Request failed with status code 404","name":"AxiosError","stack":"AxiosError: Request failed with status code 404\n    at settle (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\axios\\lib\\core\\settle.js:19:12)\n    at Unzip.handleStreamEnd (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\axios\\lib\\adapters\\http.js:599:11)\n    at Unzip.emit (node:events:531:35)\n    at Unzip.emit (node:domain:488:12)\n    at endReadableNT (node:internal/streams/readable:1696:12)\n    at processTicksAndRejections (node:internal/process/task_queues:82:21)\n    at Axios.request (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\axios\\lib\\core\\Axios.js:45:41)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async MonifyService.verifyPayment (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\payment.ts:193:24)\n    at async verifyTransaction (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:80:27)","status":404},"level":"error","message":"Payment verification failed","reference":"MNFY|57|20250201112313|00000","timestamp":"2025-02-21T17:07:47.420Z"}
{"level":"error","message":"Payment verification failed: There's no transaction matching supplied reference. Please confirm supplied reference and try again.","timestamp":"2025-02-21T17:07:47.423Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Payment verification failed: There's no transaction matching supplied reference. Please confirm supplied reference and try again.","method":"POST","path":"/api/v1/verify-payment","stack":"Error: Payment verification failed: There's no transaction matching supplied reference. Please confirm supplied reference and try again.\n    at MonifyService.verifyPayment (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\payment.ts:221:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async verifyTransaction (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:80:27)","timestamp":"2025-02-21T17:07:47.425Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T17:08:30.148Z"}
{"error":"Request failed with status code 404","level":"error","message":"Payment verification failed","reference":"MNFY|57|20250201112313|00000","timestamp":"2025-02-21T17:08:47.455Z"}
{"level":"error","message":"Payment verification failed: There's no transaction matching supplied reference. Please confirm supplied reference and try again.","timestamp":"2025-02-21T17:08:47.456Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Payment verification failed: There's no transaction matching supplied reference. Please confirm supplied reference and try again.","method":"POST","path":"/api/v1/verify-payment","stack":"Error: Payment verification failed: There's no transaction matching supplied reference. Please confirm supplied reference and try again.\n    at MonifyService.verifyPayment (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\payment.ts:221:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async verifyTransaction (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:80:27)","timestamp":"2025-02-21T17:08:47.457Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T17:09:40.146Z"}
{"error":{"code":"ERR_BAD_REQUEST","config":{"adapter":["xhr","http","fetch"],"baseURL":"https://sandbox.monnify.com","env":{},"headers":{"Accept":"application/json, text/plain, */*","Accept-Encoding":"gzip, compress, deflate, br","Authorization":"Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsibW9ubmlmeS12YWx1ZS1hZGRlZC1zZXJ2aWNlIiwibW9ubmlmeS1wYXltZW50LWVuZ2luZSIsIm1vbm5pZnktZGlzYnVyc2VtZW50LXNlcnZpY2UiLCJtb25uaWZ5LW9mZmxpbmUtcGF5bWVudC1zZXJ2aWNlIl0sInNjb3BlIjpbInByb2ZpbGUiXSwiZXhwIjoxNzQwMTYxMDE3LCJhdXRob3JpdGllcyI6WyJNUEVfTUFOQUdFX0xJTUlUX1BST0ZJTEUiLCJNUEVfVVBEQVRFX1JFU0VSVkVEX0FDQ09VTlQiLCJNUEVfSU5JVElBTElaRV9QQVlNRU5UIiwiTVBFX1JFU0VSVkVfQUNDT1VOVCIsIk1QRV9DQU5fUkVUUklFVkVfVFJBTlNBQ1RJT04iLCJNUEVfUkVUUklFVkVfUkVTRVJWRURfQUNDT1VOVCIsIk1QRV9ERUxFVEVfUkVTRVJWRURfQUNDT1VOVCIsIk1QRV9SRVRSSUVWRV9SRVNFUlZFRF9BQ0NPVU5UX1RSQU5TQUNUSU9OUyJdLCJqdGkiOiJmYWQ2ODE4ZS03NDQ1LTRkY2UtOWFjYy02NjQxMGRmNzBlZWYiLCJjbGllbnRfaWQiOiJNS19URVNUXzA2UDk5RlRONzEifQ.PrUvXTWxmQYEQsAd1oaRSRfUdhsDufBqpvVM7OEMfcR1NZisxpLns51CNx9QMoZjlRLVbNtDyN9NVBAajbZsKT0Op30IeLeJJ45dOwyaNCQoCkZtHDcbbHDVNRd7uwjjyG37cIxZvS4DDERXEVAUz_V-ONa0spRWHTmbJKyTLQyzuyfgRBTwgycyFKITbFUVC9QKkdy9HZTAKeC47zbIThyve4gLh8_jayIHnQa6tqZ6cTu0ZbCLsuaD9GywQ4yqiPwPxKlblPYOSCutYpuhPVKescrDGbVgcv0SN5Y-1tm77OjvNfPP2Euz00ZtkGo9ilbBo90PkMHgrvk8wEJvxg","Content-Type":"application/json","User-Agent":"axios/1.7.9"},"maxBodyLength":-1,"maxContentLength":-1,"method":"get","timeout":10000,"transformRequest":[null],"transformResponse":[null],"transitional":{"clarifyTimeoutError":false,"forcedJSONParsing":true,"silentJSONParsing":true},"url":"/api/v2/transactions/MNFY%7C57%7C20250201112313%7C00000","xsrfCookieName":"XSRF-TOKEN","xsrfHeaderName":"X-XSRF-TOKEN"},"message":"Request failed with status code 404","name":"AxiosError","stack":"AxiosError: Request failed with status code 404\n    at settle (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\axios\\lib\\core\\settle.js:19:12)\n    at Unzip.handleStreamEnd (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\axios\\lib\\adapters\\http.js:599:11)\n    at Unzip.emit (node:events:531:35)\n    at Unzip.emit (node:domain:488:12)\n    at endReadableNT (node:internal/streams/readable:1696:12)\n    at processTicksAndRejections (node:internal/process/task_queues:82:21)\n    at Axios.request (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\axios\\lib\\core\\Axios.js:45:41)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async MonifyService.verifyPayment (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\payment.ts:193:24)\n    at async verifyTransaction (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:80:27)","status":404},"level":"error","message":"Payment verification failed","reference":"MNFY|57|20250201112313|00000","timestamp":"2025-02-21T17:10:58.640Z"}
{"level":"error","message":"Payment verification failed: There's no transaction matching supplied reference. Please confirm supplied reference and try again.","timestamp":"2025-02-21T17:10:58.643Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Payment verification failed: There's no transaction matching supplied reference. Please confirm supplied reference and try again.","method":"POST","path":"/api/v1/verify-payment","stack":"Error: Payment verification failed: There's no transaction matching supplied reference. Please confirm supplied reference and try again.\n    at MonifyService.verifyPayment (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\services\\payment.ts:221:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async verifyTransaction (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:80:27)","timestamp":"2025-02-21T17:10:58.648Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T17:12:50.497Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T20:55:08.755Z"}
{"code":"ESOCKET","command":"CONN","host":"173.194.76.108","level":"error","message":"Email service connection failed: Client network socket disconnected before secure TLS connection was established","port":465,"stack":"Error: Client network socket disconnected before secure TLS connection was established\n    at TLSSocket.onConnectEnd (node:_tls_wrap:1727:19)\n    at TLSSocket.emit (node:events:531:35)\n    at TLSSocket.emit (node:domain:488:12)\n    at endReadableNT (node:internal/streams/readable:1696:12)\n    at processTicksAndRejections (node:internal/process/task_queues:82:21)","timestamp":"2025-02-21T20:55:28.036Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T20:55:28.039Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T21:02:11.356Z"}
{"code":"ESOCKET","command":"CONN","host":"173.194.76.108","level":"error","message":"Email service connection failed: Client network socket disconnected before secure TLS connection was established","port":465,"stack":"Error: Client network socket disconnected before secure TLS connection was established\n    at TLSSocket.onConnectEnd (node:_tls_wrap:1727:19)\n    at TLSSocket.emit (node:events:531:35)\n    at TLSSocket.emit (node:domain:488:12)\n    at endReadableNT (node:internal/streams/readable:1696:12)\n    at processTicksAndRejections (node:internal/process/task_queues:82:21)","timestamp":"2025-02-21T21:02:14.289Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T21:02:14.291Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T21:02:48.084Z"}
{"code":"ESOCKET","command":"CONN","host":"173.194.76.108","level":"error","message":"Email service connection failed: Client network socket disconnected before secure TLS connection was established","port":465,"stack":"Error: Client network socket disconnected before secure TLS connection was established\n    at TLSSocket.onConnectEnd (node:_tls_wrap:1727:19)\n    at TLSSocket.emit (node:events:531:35)\n    at TLSSocket.emit (node:domain:488:12)\n    at endReadableNT (node:internal/streams/readable:1696:12)\n    at processTicksAndRejections (node:internal/process/task_queues:82:21)","timestamp":"2025-02-21T21:02:49.163Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T21:02:49.165Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-21T23:24:13.734Z"}
{"level":"error","message":"Cannot read properties of undefined (reading 'id')","timestamp":"2025-02-22T12:59:53.612Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-22T12:59:53.614Z"}
{"level":"error","message":"Unableto update transaction","timestamp":"2025-02-22T13:28:49.707Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Unableto update transaction","method":"POST","path":"/api/v1/verify-payment","stack":"Error: Unableto update transaction\n    at verifyTransaction (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:94:15)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-02-22T13:28:49.711Z"}
{"level":"error","message":"Unable to update transaction","timestamp":"2025-02-22T13:29:37.903Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Unable to update transaction","method":"POST","path":"/api/v1/verify-payment","stack":"Error: Unable to update transaction\n    at verifyTransaction (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:95:15)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-02-22T13:29:37.905Z"}
{"level":"error","message":"Unable to update transaction","timestamp":"2025-02-22T13:31:55.596Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Unable to update transaction","method":"POST","path":"/api/v1/verify-payment","stack":"Error: Unable to update transaction\n    at verifyTransaction (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:95:15)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-02-22T13:31:55.599Z"}
{"level":"error","message":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","timestamp":"2025-02-22T13:46:17.692Z"}
{"error":{"code":11000,"errorResponse":{"code":11000,"errmsg":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"level":"error","message":"[Global Error Handler] E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","method":"POST","path":"/api/v1/make-payment","stack":"MongoServerError: E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }\n    at InsertOneOperation.execute (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\insert.ts:88:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async tryOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:278:14)\n    at async executeOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:112:12)\n    at async Collection.insertOne (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\collection.ts:284:12)","timestamp":"2025-02-22T13:46:17.733Z"}
{"error":{},"level":"error","message":"[Global Error Handler] Cannot read properties of undefined (reading 'split')","method":"POST","path":"/api/v1/make-payment","stack":"TypeError: Cannot read properties of undefined (reading 'split')\n    at authenticationToken (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\authController.ts:219:32)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\express\\lib\\router\\index.js:47:12)","timestamp":"2025-02-22T13:47:50.236Z"}
{"level":"error","message":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","timestamp":"2025-02-22T13:49:30.197Z"}
{"error":{"code":11000,"errorResponse":{"code":11000,"errmsg":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"level":"error","message":"[Global Error Handler] E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","method":"POST","path":"/api/v1/make-payment","stack":"MongoServerError: E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }\n    at InsertOneOperation.execute (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\insert.ts:88:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async tryOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:278:14)\n    at async executeOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:112:12)\n    at async Collection.insertOne (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\collection.ts:284:12)","timestamp":"2025-02-22T13:49:30.201Z"}
{"level":"error","message":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","timestamp":"2025-02-22T13:51:13.507Z"}
{"error":{"code":11000,"errorResponse":{"code":11000,"errmsg":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"level":"error","message":"[Global Error Handler] E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","method":"POST","path":"/api/v1/make-payment","stack":"MongoServerError: E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }\n    at InsertOneOperation.execute (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\insert.ts:88:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async tryOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:278:14)\n    at async executeOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:112:12)\n    at async Collection.insertOne (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\collection.ts:284:12)","timestamp":"2025-02-22T13:51:13.512Z"}
{"level":"error","message":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","timestamp":"2025-02-22T13:52:19.853Z"}
{"error":{"code":11000,"errorResponse":{"code":11000,"errmsg":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"level":"error","message":"[Global Error Handler] E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","method":"POST","path":"/api/v1/make-payment","stack":"MongoServerError: E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }\n    at InsertOneOperation.execute (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\insert.ts:88:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async tryOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:278:14)\n    at async executeOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:112:12)\n    at async Collection.insertOne (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\collection.ts:284:12)","timestamp":"2025-02-22T13:52:19.856Z"}
{"level":"error","message":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","timestamp":"2025-02-22T14:07:02.912Z"}
{"error":{"code":11000,"errorResponse":{"code":11000,"errmsg":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"level":"error","message":"[Global Error Handler] E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","method":"POST","path":"/api/v1/make-payment","stack":"MongoServerError: E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }\n    at InsertOneOperation.execute (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\insert.ts:88:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async tryOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:278:14)\n    at async executeOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:112:12)\n    at async Collection.insertOne (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\collection.ts:284:12)","timestamp":"2025-02-22T14:07:02.953Z"}
{"level":"error","message":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","timestamp":"2025-02-22T14:39:25.351Z"}
{"error":{"code":11000,"errorResponse":{"code":11000,"errmsg":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"level":"error","message":"[Global Error Handler] E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","method":"POST","path":"/api/v1/make-payment","stack":"MongoServerError: E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }\n    at InsertOneOperation.execute (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\insert.ts:88:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async tryOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:278:14)\n    at async executeOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:112:12)\n    at async Collection.insertOne (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\collection.ts:284:12)","timestamp":"2025-02-22T14:39:25.394Z"}
{"level":"error","message":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","timestamp":"2025-02-22T14:40:45.991Z"}
{"error":{"code":11000,"errorResponse":{"code":11000,"errmsg":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"level":"error","message":"[Global Error Handler] E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","method":"POST","path":"/api/v1/make-payment","stack":"MongoServerError: E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }\n    at InsertOneOperation.execute (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\insert.ts:88:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async tryOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:278:14)\n    at async executeOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:112:12)\n    at async Collection.insertOne (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\collection.ts:284:12)","timestamp":"2025-02-22T14:40:46.013Z"}
{"level":"error","message":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","timestamp":"2025-02-22T14:42:18.393Z"}
{"error":{"code":11000,"errorResponse":{"code":11000,"errmsg":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"level":"error","message":"[Global Error Handler] E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","method":"POST","path":"/api/v1/make-payment","stack":"MongoServerError: E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }\n    at InsertOneOperation.execute (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\insert.ts:88:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async tryOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:278:14)\n    at async executeOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:112:12)\n    at async Collection.insertOne (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\collection.ts:284:12)","timestamp":"2025-02-22T14:42:18.414Z"}
{"level":"error","message":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","timestamp":"2025-02-22T14:43:44.411Z"}
{"error":{"code":11000,"errorResponse":{"code":11000,"errmsg":"E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"index":0,"keyPattern":{"reference":1},"keyValue":{"reference":null}},"level":"error","message":"[Global Error Handler] E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }","method":"POST","path":"/api/v1/make-payment","stack":"MongoServerError: E11000 duplicate key error collection: Utility.transactions index: reference_1 dup key: { reference: null }\n    at InsertOneOperation.execute (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\insert.ts:88:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async tryOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:278:14)\n    at async executeOperation (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\operations\\execute_operation.ts:112:12)\n    at async Collection.insertOne (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\node_modules\\mongodb\\src\\collection.ts:284:12)","timestamp":"2025-02-22T14:43:44.432Z"}
{"level":"error","message":"Unable to update transaction","timestamp":"2025-02-22T14:51:05.471Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Unable to update transaction","method":"POST","path":"/api/v1/verify-payment","stack":"Error: Unable to update transaction\n    at verifyTransaction (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:107:15)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-02-22T14:51:05.479Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-22T20:38:32.443Z"}
{"code":"EDNS","command":"CONN","errno":-3008,"hostname":"smtp.gmail.com","level":"error","message":"Email service connection failed: getaddrinfo ENOTFOUND smtp.gmail.com","stack":"Error: getaddrinfo ENOTFOUND smtp.gmail.com\n    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:118:26)","syscall":"getaddrinfo","timestamp":"2025-02-22T20:38:32.491Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-22T20:38:32.492Z"}
{"level":"error","message":"Unable to update transaction","timestamp":"2025-02-22T20:38:53.426Z"}
{"error":{"isOperational":true,"statusCode":500},"level":"error","message":"[Global Error Handler] Unable to update transaction","method":"POST","path":"/api/v1/verify-payment","stack":"Error: Unable to update transaction\n    at verifyTransaction (C:\\Users\\FOLTI-ADMIN\\Documents\\Ambitious_data\\VTU\\src\\controllers\\paymentController.ts:107:15)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-02-22T20:38:53.429Z"}
{"level":"error","message":"Unable to update transaction","timestamp":"2025-02-22T20:41:14.006Z"}
{"level":"error","message":"Transaction not paid","timestamp":"2025-02-22T20:44:53.940Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-23T13:36:24.119Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-23T13:50:06.762Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-23T13:50:28.701Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-23T13:50:59.350Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-23T13:52:06.299Z"}
{"level":"error","message":"Unhandled Rejection at:","timestamp":"2025-02-23T15:54:51.595Z"}
